remote-config/src/fetch/fetcher.rs

```diff
@@ -0,0 +1,316 @@
+    ///  - removes unused files
+    ///  - checks if the files are already known,
+    ///  - stores new files,
+    ///  - returns all currently active files.
+    /// It also makes sure that old files are dropped before new files are inserted.
+    pub async fn fetch_once(
+        &mut self,
+        runtime_id: &str,
+        target: Arc<Target>,
+        config_id: &str,
+        last_error: Option<String>,
+        opaque_state: &mut OpaqueState,
+    ) -> anyhow::Result<Vec<Arc<S::StoredFile>>> {
+        let Target { service, env, app_version } = (*target).clone();
+
+        let mut cached_target_files = vec![];
+        let mut config_states = vec![];
+
+        for StoredTargetFile { state, meta, .. } in self.state.target_files_by_path.lock().unwrap().values() {
+            config_states.push(state.clone());
+            cached_target_files.push(meta.clone());
+        }
+
+        let config_req = ClientGetConfigsRequest {
+            client: Some(datadog_trace_protobuf::remoteconfig::Client {
+                state: Some(ClientState {
+                    root_version: 1,
+                    targets_version: 0,
+                    config_states,
+                    has_error: last_error.is_some(),
+                    error: last_error.unwrap_or_default(),
+                    backend_client_state: std::mem::take(&mut opaque_state.client_state),
+                }),
+                id: config_id.into(),
+                // TODO maybe not hardcode requested products?
+                products: vec!["APM_TRACING".to_string(), "LIVE_DEBUGGING".to_string()],
+                is_tracer: true,
+                client_tracer: Some(ClientTracer {
+                    runtime_id: runtime_id.to_string(),
+                    language: self.state.invariants.language.to_string(),
+                    tracer_version: self.state.invariants.tracer_version.clone(),
+                    service,
+                    extra_services: vec![],
+                    env,
+                    app_version,
+                    tags: vec![],
+                }),
+                is_agent: false,
+                client_agent: None,
+                last_seen: 0,
+                capabilities: vec![],
+            }),
+            cached_target_files,
+        };
+        let json = serde_json::to_string(&config_req)?;
+
+        // TODO: directly talking to datadog endpoint (once signatures are validated)
+        let req = self.state.endpoint
+            .into_request_builder(concat!("Sidecar/", env!("CARGO_PKG_VERSION")))?;
+        let response = Client::builder()
+            .build(connector::Connector::default())
+            .request(req.body(Body::from(json))?)
+            .await
+            .map_err(|e| anyhow::Error::msg(e).context(format!("Url: {:?}", self.state.endpoint)))?;
+        let status = response.status();
+        let body_bytes = hyper::body::to_bytes(response.into_body()).await?;
+        if status != StatusCode::OK {
```

Comment:
It has have to check the error code better, this is going to spam the logs if RC is disabled. The system polling for configurations need to know it should not retry and that it's not an error (just that RC is not enabled)

Reply:
Good point, how can I distinguish error from disabled? Do you know which status code or such that is then?

Reply:
I am comparing (where possible of course) to with https://github.com/DataDog/dd-trace-py/blob/fa50018333f317ffac9b1c75464080a3cf5361f7/ddtrace/internal/remoteconfig/client.py

And the handling is here
https://github.com/DataDog/dd-trace-py/blob/fa50018333f317ffac9b1c75464080a3cf5361f7/ddtrace/internal/remoteconfig/client.py#L325

Empty reply => ok, no change

code 404 => not enabled or unsupported
the rest is as usual

Reply:
Thanks, I'll update the implementation accordingly.

Reply:
> The system polling for configurations need to know it should not retry

@mellon85 Do we need to handle the case when someone changes agent config? Or do you mean avoiding extra retries which are not part of the 5-second refresh rate?

Reply:
I think the handling is correct now, with returning an Option::None when empty.